### DISCUSSION TITLE
Building project with xelatex

### DISCUSSION THREAD

**Discussion by bhaddow** [James-Yu/LaTeX-Workshop#3130] [Q&A]
**Is your feature request related to a problem? Please describe.**

I would like to be able to build with xlatex, but it does not appear within the default recipes. I have searched the documentation and the user interface, but I cannot see how to add a new recipe. Is it possible?

**Describe the solution you'd like**

I would like to be able to add a new recipe, which builds with xelatex.

**Describe alternatives you've considered**

**Additional context**

**Comment by seragunn** [comment]
In `latex-workshop.latex.tools` you can edit `latexmk` to
```
{
    "name": "latexmk",
    "command": "latexmk",
    "args": [
      "-synctex=1",
      "-interaction=nonstopmode",
      "-file-line-error",
      "-xelatex",
      "-outdir=%OUTDIR%",
      "%DOC%"
    ],
    "env": {}
},
```

(I.e. replacing `-pdf` with `-xelatex`).

Alternatively, you could make a new tool called `xelatex` by copying the `pdflatex` tool (again in `latex-workshop.latex.tools`). And then copy the `pdflatex ➞ bibtex ➞ pdflatex`×2` recipe. So something like:

```
# latex-workshop.latex.tools
{
    "name": "xelatex",
    "command": "xelatex",
    "args": [
      "-synctex=1",
      "-interaction=nonstopmode",
      "-file-line-error",
      "%DOC%"
    ],
    "env": {}
}
# latex-workshop.latex.recipes
  {
    "name": "xelatex ➞ bibtex ➞ xelatex`×2",
    "tools": [
      "xelatex",
      "bibtex",
      "xelatex",
      "xelatex"
    ]
  }
```
(and if you don't need bibtex you can remove that step).

But basically you need to create a tool before you create the recipe.

**Comment by tamuratak** [comment]
You can also refer to https://github.com/James-Yu/LaTeX-Workshop/wiki/Compile#latex-recipes.

**Comment by bhaddow** [comment]
This is probably a silly question, but what do you mean by this?

> In latex-workshop.latex.tools you can edit latexmk to

Is there a file somewhere called `latex-workshop.latex.tools ` ?

**Comment by tamuratak** [comment]
See

- https://github.com/James-Yu/LaTeX-Workshop/wiki/Install#settings
- https://github.com/James-Yu/LaTeX-Workshop/wiki/Compile#latex-recipes

**Comment by bhaddow** [comment]
Yes, but where are the recipes stored?

**Comment by tamuratak** [comment]
settings.json

Please read documents before asking.

**Comment by bhaddow** [comment]
In .vscode , inside the current project? I tried adding recipes there, and they did not appear in the menu

**Comment by bhaddow** [comment]
Note that the document does not mention settings.json

**Comment by tamuratak** [comment]
We cannot help you.
